flintrock/ec2.py

```diff
@@ -334,9 +348,12 @@ def get_or_create_ec2_security_groups(
             VpcId=vpc_id)
 
     # Rules for the client interacting with the cluster.
-    flintrock_client_ip = (
-        urllib.request.urlopen('http://checkip.amazonaws.com/')
-        .read().decode('utf-8').strip())
+    if use_private_vpc:
```

Comment:
I've reviewed the relevant docs here:
* http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_Introduction.html
* http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/vpc-nat.html

It sounds like your VPC has an internet gateway attached, otherwise the call to `checkip` would fail, or perhaps would return a private address.

Even with the gateway attached, from my reading of the docs, it sounds like only instances from the same VPC can ever get the same public IP address. So I don't think it's an issue to authorize both addresses.